### PR TITLE
Symbol.species tests

### PR DESCRIPTION
**Pull request by smikes**
Initial tests for property existence
move getter name test to builtin-getter-name
initial cut at extends test

### CODE REVIEW

File: test/built-ins/Symbol/species/Symbol.species.in_Array.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2015 Cubane Canada, Inc.  All rights reserved.
+// See LICENSE for details.
+
+/*---
+info: >
+ Array has a property at `Symbol.species`
+bes6id: 22.1.2.5
```
Comment (anba) [inline test/built-ins/Symbol/species/Symbol.species.in_Array.js]: Typo "bes6id"

File: test/built-ins/Symbol/species/Symbol.species.builtin-getter-name.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ ES6 spec 'get [Symbol.species]'
+es6id: 21.2.4.2, 22.1.2.5, 22.2.2.4, 23.1.2.2, 23.2.2.2
+author: Sam Mikes
+description: Symbol.species getters have defined names
+includes: 
+  - propertyHelper.js
+---*/
+
+function getGetterName(obj, name) {
+    var getter = Object.getOwnPropertyDescriptor(obj, Symbol.species).get.name;
```
Comment (anba) [inline test/built-ins/Symbol/species/Symbol.species.builtin-getter-name.js]: Trailing `.name` needs to be removed.

File: test/built-ins/Symbol/species/Symbol.species.notChangedByExtends.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  - propertyHelper.js
+---*/
+
+class MyRegExp extends RegExp {
+};
+
+assert.sameValue(MyRegExp[Symbol.species], MyRegExp);
+
+assert.sameValue(
+    Object.getOwnPropertyDescriptor(MyRegExp, Symbol.species).get,
```
Comment (anba) [inline test/built-ins/Symbol/species/Symbol.species.notChangedByExtends.js]: `MyRegExp` does not have `Symbol.species` own property.

File: test/built-ins/Symbol/species/Symbol.species.in_ArrayBuffer.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2015 Cubane Canada, Inc.  All rights reserved.
+// See LICENSE for details.
+
+/*---
+info: >
+ ArrayBuffer has a property at `Symbol.species`
+bes6id: 24.1.3.3
```
Comment (anba) [inline test/built-ins/Symbol/species/Symbol.species.in_ArrayBuffer.js]: Typo "bes6id"

File: test/built-ins/Symbol/species/Symbol.species.builtin-getter-name.js (hunk 2 above)
Comment (smikes) [inline test/built-ins/Symbol/species/Symbol.species.builtin-getter-name.js]: Thanks!

File: test/built-ins/Symbol/species/Symbol.species.notChangedByExtends.js (hunk 3 above)
Comment (smikes) [inline test/built-ins/Symbol/species/Symbol.species.notChangedByExtends.js]: Thanks!

**Comment by smikes** [comment]
Incorporated line notes and rebased.  Additional review (@bterlson) ?

**Comment by bterlson** [comment]
Sorry for the delay, I've been out of office and then preparing for some upcoming talks... I'm reviewing this now :)

File: test/built-ins/Symbol/species/Symbol.species.builtin-getter-name.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2015 Cubane Canada, Inc.  All rights reserved.
+// See LICENSE for details.
+
+/*---
+info: >
+ ES6 spec 'get [Symbol.species]'
+es6id: 21.2.4.2, 22.1.2.5, 22.2.2.4, 23.1.2.2, 23.2.2.2
```
Comment (bterlson) [inline test/built-ins/Symbol/species/Symbol.species.builtin-getter-name.js]: Interesting, I think this is the first I've seen a list for es6id :)

File: test/built-ins/Symbol/species/Symbol.species.in_Map.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2015 Cubane Canada, Inc.  All rights reserved.
+// See LICENSE for details.
+
+/*---
+info: >
+ Map has a property at `Symbol.species`
+es6id: 23.1.2.2
+author: Sam Mikes
+description: Map[Symbol.species] exists per spec
```
Comment (bterlson) [inline test/built-ins/Symbol/species/Symbol.species.in_Map.js]: Probably could stand to add the features array for this one so implementations without map can skip (although not super important)

File: test/built-ins/Symbol/species/Symbol.species.notChangedByExtends.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+author: Sam Mikes
+description: Symbol.species is retained on subclassing
+includes: 
+  - propertyHelper.js
+---*/
+
+class MyRegExp extends RegExp {
+};
+
+assert.sameValue(MyRegExp[Symbol.species], MyRegExp);
```
Comment (bterlson) [inline test/built-ins/Symbol/species/Symbol.species.notChangedByExtends.js]: I think this is just normal prototype lookup? But I suppose this is a good happy path test case regardless.

**Comment by rwaldron** [comment]
Once this is landed, I'll have the V8 Symbol tests ready to go. https://github.com/bocoup/test262/commits/harmony-symbols 

(There are a few remaining items to address)

**Comment by bterlson** [comment]
Since you're dependent on this I'll just pull it in for now. We can fill in the features array later if necessary.

**Comment by smikes** [comment]
I am still on the road but will be back next week.  I will address the line notes with a new PR.

Thanks!

**Pull request merged**